token-metadata/program/src/instruction.rs

```diff
@@ -12,27 +12,37 @@ use solana_program::{
     pubkey::Pubkey,
     sysvar,
 };
+#[cfg(feature = "serde")]
+use {
+    serde::{Deserialize, Serialize},
+    serde_with::{As, DisplayFromStr},
+};
 
 #[repr(C)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(BorshSerialize, BorshDeserialize, PartialEq, Debug, Clone)]
 /// Args for update call
 pub struct UpdateMetadataAccountArgs {
     pub data: Option<Data>,
+    #[cfg_attr(feature = "serde", serde(with = "As::<Option<DisplayFromStr>>"))]
```

Comment:
Is this syntax required since serde is gated as a crate feature, otherwise it could just be `as = DisplayFromStr`?

Reply:
Yes, serde_with doesn't work as is with features, see: https://docs.rs/serde_with/1.10.0/serde_with/guide/serde_as/index.html#gating-serde_as-on-features